src/Router.js

```diff
@@ -33,12 +33,19 @@ export default class Router {
       base: '',
     }, opts)
 
-    this.setInitialState();
-
     this.onRouteStart = this.opts.onRouteStart;
-    this.onRouteComplete = this.opts.onRouteComplete
+    this.onRouteComplete = this.opts.onRouteComplete;
+    this.__onpopstate = this.__onpopstate.bind(this);
+  }
 
-    WindowEnv.addEventListener('popstate', this.__onpopstate.bind(this))
```

Comment:
@zachpwr This was the previous method to add the event listener. Using `bind` in this way prevents the `removeEventListener` call from having any effect because `this.__onpopstate` is not set to the return value of `.bind` here.

I moved the `bind` logic to the constructor to solve this.